src/vm/prestub.cpp

```diff
@@ -1976,14 +1957,12 @@ PCODE MethodDesc::DoPrestub(MethodTable *pDispatchingMT)
     }
 
     /**************************   BACKPATCHING   *************************/
-    // See if the addr of code has changed from the pre-stub
-
 #ifdef FEATURE_CODE_VERSIONING
-    if (IsVersionableWithoutJumpStamp())
+    if (IsVersionable())
```

Comment:
IsVersionableWithoutJumpStamp() was previously a single bit check and now IsVersionable() aggregates a variety of different checks. I think @kouvel had the bit cached because the checks were being called frequently enough that the expense of the uncached check became noticeable. When you profile the new code you and @kouvel may want to keep an eye if the expense of IsVersionable() merits any change in the caching strategy.

Reply:
Also for calls to `IsVersionableWith*()` functions. Now for methods that are not eligible for tiered compilation it would also evaluate `IsEligibleForReJIT()`. I think `ReJitManager::IsReJITEnabled()` could be made faster by caching the configured value for `CLRConfig::EXTERNAL_ProfAPI_RejitOnAttach` in `EEConfig` as done for other config vars. That would turn `IsEligibleForReJIT()` into just a few checks on memory, which hopefully won't be noticeable.

Reply:
May be possible to turn `IsReJITEnabled()` into a single check. `IsEligibleForTieredCompilation()` is called directly in many places, so it may still be beneficial to continue having the cached bit represent that instead of `IsVersionable()`.

Reply:
Actually looks like `EXTERNAL_ProfAPI_RejitOnAttach` is already cached in a static local variable, so it may be ok already

Reply:
Both of the checks in `ReJitManager::IsReJITEnabled` are cached in local statics, so I don't believe there should be a perf concern here. Doing the perf testing will prove that right or wrong.